ldes-server-domain/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/domain/ldesfragment/services/RootFragmentCreatorImpl.java

```diff
@@ -18,16 +17,13 @@ public RootFragmentCreatorImpl(LdesFragmentRepository ldesFragmentRepository) {
 
 	@Override
 	public LdesFragment createRootFragmentForView(String viewName) {
-		Optional<LdesFragment> optionalRoot = ldesFragmentRepository
-				.retrieveRootFragment(viewName);
-		if (optionalRoot.isEmpty()) {
-			return createRoot(viewName);
-		}
-		return optionalRoot.get();
+		return ldesFragmentRepository
+				.retrieveRootFragment(viewName)
+				.orElseGet(() -> createRoot(viewName));
 	}
 
 	private LdesFragment createRoot(String viewName) {
-		LdesFragment ldesFragment = new LdesFragment(viewName, List.of());
+		LdesFragment ldesFragment = new LdesFragment(viewName.split("/")[0], viewName, List.of());
```

Comment:
Would avoid this split and just create a getter on ViewSpecification for the attribute collectionName.